contracts/src/spaces/facets/roles/RolesBase.sol

```diff
@@ -318,6 +324,118 @@ abstract contract RolesBase is IRolesBase {
     emit RoleRemoved(msg.sender, roleId);
   }
 
+  // =============================================================
+  // Channel Permissions
+  // =============================================================
+  function _getCustomChannelPermissions(
+    uint256 roleId,
+    bytes32 channelId
+  ) internal view returns (string[] memory permissions) {
+    // check role exists
+    _checkRoleExists(roleId);
+
+    // check channel exists
+    ChannelService.checkChannelExists(channelId);
+
+    return
+      RolesStorage
+      .layout()
+      .permissionByChannelIdByRoleId[roleId][channelId].values();
+  }
+
+  function _createCustomChannelPermissions(
+    uint256 roleId,
+    bytes32 channelId,
+    string[] memory permissions
+  ) internal {
+    ChannelService.checkChannelExists(channelId);
+
+    // check role exists
+    _checkRoleExists(roleId);
+
+    RolesStorage.Layout storage rs = RolesStorage.layout();
+
+    rs.channelsByRole[roleId].add(channelId);
+
+    // check permissions is not empty
+    uint256 permissionsLen = permissions.length;
+    for (uint256 i = 0; i < permissionsLen; i++) {
+      _checkEmptyString(permissions[i]);
+      if (rs.roleById[roleId].permissions.contains(permissions[i])) {
```

Comment:
This condition is confusing - sorry if I misunderstood! Are you checking that the default permissions completely don't overlap with override permissions?

Reply:
thanks for the catch!